src/reload.go

```diff
@@ -263,52 +287,69 @@ func nginxPlus(data *RenderingData) error {
 			"nginx": config.Nginxplusapiaddr,
 		}).Info("endpoint")
 
-		endpoint := "http://" + config.Nginxplusapiaddr + "/api"
-
-		tr := &http.Transport{
-			MaxIdleConns:       30,
-			DisableCompression: true,
-		}
-
-		client := &http.Client{Transport: tr}
-		c := NginxClient{endpoint, client}
-		nginxClient, error := NewNginxClient(c.httpClient, c.apiEndpoint)
-		if error != nil {
-			logger.WithFields(logrus.Fields{
-				"error": error,
-			}).Error("unable to make call to nginx plus")
-			return error
-		}
 		upstreamtocheck := app.Vhost
-		var finalformattedServers []UpstreamServer
+		var finalformattedServers []nplus.UpstreamServer
 
 		for _, server := range newFormattedServers {
-			formattedServer := UpstreamServer{Server: server, MaxFails: config.MaxFailsUpstream, FailTimeout: config.FailTimeoutUpstream, SlowStart: config.SlowStartUpstream}
+			formattedServer := nplus.UpstreamServer{Server: server, MaxFails: config.MaxFailsUpstream, FailTimeout: config.FailTimeoutUpstream, SlowStart: config.SlowStartUpstream}
 			finalformattedServers = append(finalformattedServers, formattedServer)
 		}
+		err := nginxClient.CheckIfUpstreamExists(upstreamtocheck)
```

Comment:
Please add the following comment as per the original code
`// If upstream has no servers, UpdateHTTPServers returns error as in-line GetHTTPServers returns error. server ID 0 needs to be explicitly initiated by a PATCH`

Reply:
Done